src/components/payment/Final.jsx

```diff
@@ -0,0 +1,88 @@
+import { useState, useContext } from 'react';
+import { AuthContext } from '../../context/AuthContext';
+import { PaymentCont, SubmissionForm, UPI } from './Payment';
+import { Heading, Paragraph } from '../shared';
+import { Contactorg, Emaildata } from '../../data/paymentData';
+import { uploadReceipt } from './uploadReceipt';
+import { updateDocumentByUid } from '../../firebase/uploadProof';
+
+function Final() {
+	const { userInfo } = useContext(AuthContext);
+	var currentUser = userInfo[0];
```

Comment:
nit: can be const as is not being changed I believe.

Furthermore please show some error message using toast and redirect if user is not logged in.